### ISSUE TITLE
[Android] App crashes on android 4.x (API 18, 19)

### ISSUE THREAD

**Issue by gaykov** [Microsoft/react-native-code-push#1026]
Application (it's just a simple app with codepush integration)  crashes if I'm trying to run it on Android 4.3 or 4.3 and the same application works fine on all higher android versions.

### Steps to Reproduce
1. Run an application on a simulator with android 4.3 or 4.4 

### Actual Behavior
Application crashes right after the start with the following stack trace:

```
E/AndroidRuntime: FATAL EXCEPTION: main
                  java.lang.NoClassDefFoundError: com.microsoft.codepush.react.CodePushUpdateManager
                      at com.microsoft.codepush.react.CodePush.<init>(CodePush.java:63)
                      at de.ex.app.MainApplication$1.getPackages(MainApplication.java:35)
                      at com.facebook.react.ReactNativeHost.createReactInstanceManager(ReactNativeHost.java:74)
                      at com.facebook.react.ReactNativeHost.getReactInstanceManager(ReactNativeHost.java:41)
                      at com.facebook.react.ReactActivityDelegate.loadApp(ReactActivityDelegate.java:111)
                      at com.facebook.react.ReactActivityDelegate.onCreate(ReactActivityDelegate.java:100)
                      at com.facebook.react.ReactActivity.onCreate(ReactActivity.java:54)
                      at android.app.Activity.performCreate(Activity.java:5133)
                      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2175)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2261)
                      at android.app.ActivityThread.access$600(ActivityThread.java:141)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
                      at android.os.Handler.dispatchMessage(Handler.java:99)
                      at android.os.Looper.loop(Looper.java:137)
                      at android.app.ActivityThread.main(ActivityThread.java:5103)
                      at java.lang.reflect.Method.invokeNative(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:525)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
                      at dalvik.system.NativeStart.main(Native Method)

```

### Environment

* react-native-code-push version: 5.1.0-beta
* react-native version: 0.46.4
* iOS/Android/Windows version: Android 4.3 and 4.4
* Does this reproduce on a debug build or release build? Debug build
* Does this reproduce on a simulator, or only on a physical device? Tested on simulator only

**Comment by sergey-akhalkov** [comment]
Hi @gaykov, thanks for reaching us. I've just successfully run an CodePushified React Native app on Android 4.3 - have no issues. Could you please try to install the lates version of react-native-code-push@5.1.2 because we've fixed dex overflow exception there. Please let me know if it helps

**Comment by gaykov** [comment]
@sergey-akhalkov thank you for your response! Yes, the error was fixed by upgrading react-native-code-push version to 5.1.2.

**Comment by matthiasleitner** [comment]
@sergey-akhalkov  I am seeing the same issue using 5.1.3 on an Android 4.4.2 device. Any suggestions?

**Comment by YeshanJay** [comment]
I am getting the same issue on Android `4.1.2` and `4.4.2`, but works fine on Android `8.1.0`.

Used `react-native-code-push@5.2.2`
Then I upgraded to `react-native-code-push@5.3.2`, and the issue still existed.

Also note that I have enabled multi-dex. Is that the reason? Because before enabling this, it used to work as i can remember.

**Environment**
OS - Android
RN - 0.55.2

buildToolsVersion -  23.0.1
minSdkVersion - 16
targetSdkVersion - 26
compileSdkVersion - 26

multiDexEnabled - true
dexOptions - { javaMaxHeapSize "4g" }
